tests/src/python/test_provider_memory.py

```diff
@@ -73,6 +73,10 @@ def testPointCtor(self):
         layer = QgsVectorLayer("Point", "test", "memory")
         assert layer.isValid(), "Failed to create valid point memory layer"
 
+    def testNoGeometryCtor(self):
+        layer = QgsVectorLayer("None", "test", "memory")
+        assert layer.isValid(), "Failed to create valid geometryless memory layer"
```

Comment:
Would be a good idea to check the geometry type on the layer here (and in line 74) too